### PR TITLE
Clarify deprecation message for FileTrees as outputs

### PR DESCRIPTION
**Pull request by wolfs**
The userguide does not say what a directory tree is.

### CODE REVIEW

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy [hunk 1]
```diff
@@ -1252,7 +1252,7 @@ task generate(type: TransformerTask) {
 
         then:
         skippedTasks.contains(':myTask')
-        output.contains('Adding file trees which are not directory trees as output files has been deprecated and is scheduled to be removed in Gradle 5.0')
+        output.contains('Adding file trees which are not directory based as output files has been deprecated and is scheduled to be removed in Gradle 5.0')
```
Comment (pledbrook) [inline subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy]: I would rephrase as something like:

> The ability to add non-directory-based file trees as defined outputs has been deprecated and is scheduled to be removed in Gradle 5.0

I'm not sure about "defined outputs", but that's the terminology I think I used in the section of the user manual on incremental build.

Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy]: I would use "declared outputs", at least that is the term we are using internally. In the user guide it seems that we currently use both: https://docs.gradle.org/current/userguide/more_about_tasks.html

I guess the "define" parts are from your changes and the "declare" are from me...

**Comment by wolfs** [comment]
@pledbrook I adjusted the deprecation warning, please take another look.
